src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -36,4 +84,73 @@ protected function getExtension()
     {
         return 'yml';
     }
+
+    private function expandToTree(array $messages)
+    {
+        $tree = array();
+
+        foreach ($messages as $id => $value) {
+            $this->expandElement($tree, explode('.', $id), $value);
```

Comment:
@aitboudad if you use the original message as translation id, you should not use the tree dumping. The tree dumping is meant to be used only for translation keys.